.github/workflows/workflow_tests.yml

```diff
@@ -23,13 +44,79 @@ jobs:
           architecture: 'x64'
 
       - name: Display Python version
-        run: python -c "import sys; print(sys.version)"
+        run: |
+          python -c "import sys; print(sys.version)"
+          bash -c "ls"
 
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
+          pip install .[cpu] --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/cpu/us/ -f https://download.pytorch.org/whl/torch_stable.html
+          # Dynamic link oneCCL and Intel MPI libraries
+          source $(python -c "import oneccl_bindings_for_pytorch as torch_ccl;print(torch_ccl.cwd)")/env/setvars.sh
           pip install -r ./tests/requirements.txt
 
+      - name: Start Ray Cluster
+        run: |
+          bash -c "ray start --head"
+
       - name: Start tests
         run: |
           bash -c "./tests/run-tests.sh"
+
+  build_docker:
+
+    name: docker-test
+
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      
+      - name: Determine Target
+        id: "target"
+        run: |
+          target="inference"
+          echo "target is ${target}"
+          echo "target=$target" >> $GITHUB_OUTPUT
+
+      - name: Build Docker Image
+        run: |
+          DF_SUFFIX=".cpu_and_deepspeed_test"
+          TARGET=${{steps.target.outputs.target}}
+          docker build ./ --build-arg CACHEBUST=1 -f dev/docker/Dockerfile${DF_SUFFIX} -t ${TARGET}:latest && yes | docker container prune && yes
```

Comment:
Could we move all docker related blocks into functions in some bash script and source from them? It can reduce code duplications.